### ISSUE TITLE
[Research] Integrate REVE Foundation Model for 4-Channel EEG Emotion Classification

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#19]
## What

Integrate the **REVE foundation model** (NeurIPS 2025) as an alternative/replacement for the current LightGBM feature-based emotion pipeline. REVE is pretrained on 60,000+ hours of EEG from 25,000 subjects and has a **4D positional encoding** that natively handles arbitrary electrode counts — including Muse 2's 4-channel AF7/AF8/TP9/TP10 layout.

## Why

- Current pipeline: feature-based heuristics → ~60-74% accuracy (cross-subject)
- REVE achieves SOTA on 10 downstream tasks with **+2.5% avg gain** over CBraMod (next best foundation model)
- Linear probing alone yields **up to 17% improvement** over baselines
- The 4D positional encoding explicitly handles non-standard electrode arrangements — no channel interpolation needed
- Pretrained weights are open-source on HuggingFace: `brain-bzh/reve-base`

### Step 1: Download and test REVE locally
```bash
pip install torch transformers
python -c "from huggingface_hub import snapshot_download; snapshot_download('brain-bzh/reve-base', local_dir='ml/models/saved/reve-base')"
```

### Step 2: Create adapter for Muse 2 electrode positions
```python
# ml/models/reve_adapter.py
# Map Muse 2 electrodes to REVE's 4D positional encoding:
# TP9 → left temporal position
# AF7 → left frontal position  
# AF8 → right frontal position
# TP10 → right temporal position
```

### Step 3: Fine-tune on emotion data
- Use EMOGNITION dataset (43 subjects, Muse 2 native, 9 emotions) if available
- Alternatively fine-tune on SEED/DEAP with 4-channel subset
- Two-stage: (1) freeze encoder + linear probe, (2) unfreeze + full fine-tune

### Step 4: Wire into inference chain
- Add REVE as highest-priority path in `emotion_classifier.py` predict() chain
- Gate behind accuracy threshold (≥60% CV) like existing models
- Fall back to LightGBM → feature heuristics if REVE fails

### Files to Modify
- `ml/models/reve_adapter.py` (NEW) — REVE wrapper with Muse 2 electrode mapping
- `ml/models/emotion_classifier.py` — Add REVE to prediction chain
- `ml/requirements.txt` — Add torch, transformers, huggingface_hub
- `ml/training/train_reve_emotion.py` (NEW) — Fine-tuning script

### Expected Impact
- Cross-subject emotion accuracy: **74% → 85-89%** (estimated)
- Particularly strong on valence (REVE attends to prefrontal/temporal channels)

## Complexity
HIGH — PyTorch, HuggingFace weights, custom fine-tuning pipeline

## References
- Paper: arXiv 2510.21585 (NeurIPS 2025)
- Weights: https://huggingface.co/brain-bzh/reve-base
- EEG-FM-Bench benchmark (2025): https://arxiv.org/abs/2508.17742
